libs/particles/effects.ts

```diff
@@ -21,12 +21,44 @@ namespace particles {
             if (!this.sourceFactory) return;
             this.sourceFactory(anchor, particlesPerSecond);
         }
+
+        /**
+         * Destroy the provided sprite with
+         * @param sprite
+         * @param particlesPerSecond
+         * @param lifespan how long the sprite will remain on the screen
+         */
+        //% blockId=particlesDestroySpriteWithAnimation block="use %effect effect to destroy %anchor=variables_get(mySprite) at rate %particlesPerSecond p/s || with lifespan %lifespan"
+        //% particlesPerSecond.defl=20
+        //% particlesPerSecond.min=1 particlePerSeconds.max=100
+        //% lifespan.defl=500
+        //% group="Effects"
+        destroy(anchor: Sprite, particlesPerSecond: number, lifespan: number = 500) {
+            anchor.setFlag(SpriteFlag.Ghost, true);
+            this.sourceFactory(anchor, particlesPerSecond);
+            anchor.lifespan = lifespan;
+            
+            // clear out some pixels at random to give feedback on sprite being destroyed
+            const replacementImage = anchor.image.clone();
```

Comment:
Should that be under particle effects? It wouldn't emit any particles so it feels a bit off to count as one to me; maybe a ``SpriteEffect`` class that has a handful of effects that can be applied to sprites (random clear pixels from this, invert colors, one layer of the melt effect that occurs on for game over, etc). I'm probably just overthinking that though

side question, I think I remember you mentioning that icons could be made for the different effects so they show up nicely instead of just a text drop down, but I don't remember how; something along the lines of making jres images like what is done for gallery sprites?